packages/ckeditor5-table/src/converters/tableproperties.ts

```diff
@@ -219,6 +229,45 @@ export function upcastBorderStyles(
 			conversionApi.writer.setAttribute( modelAttributes.width, reducedBorder.width, modelElement );
 		}
 	} ) );
+
+	if ( editor.config.get( 'experimentalFlags.upcastTableBorderZeroAttributes' ) ) {
+		// If parent table has `border="0"` attribute then set border style to `none`
+		// all table cells of that table and table itself.
+		conversion.for( 'upcast' ).add( dispatcher => {
+			dispatcher.on<UpcastElementEvent>( `element:${ viewElementName }`, ( evt, data, conversionApi ) => {
+				const { modelRange, viewItem } = data;
+
+				const viewTable = (
+					viewItem.is( 'element', 'table' ) ?
+						viewItem :
+						viewItem.findAncestor( 'table' )
+				)!;
+
+				// If something already consumed the border attribute on the nearest table element, skip the conversion.
+				if ( !conversionApi.consumable.test( viewTable, { attributes: 'border' } ) ) {
+					return;
+				}
```

Comment:
How does this affect setting `border:none` on both cells and the table itself? Doesn't it set it only to the first cell?

Reply:
It works fine, `upcastBorderStyles` is used for `td` and `th` elements, and consume of `border` attribute is performed after upcasting of table children. Checked there: https://github.com/ckeditor/ckeditor5/pull/19313/files#diff-e478b75c9ed00d13199f5eeab0e5efb93e27edabf2978d6ad6fe04ac25e01fa6R318